### ISSUE TITLE
Mute Functionality

### ISSUE THREAD

**Issue by dquick-freelance** [appdotnet/api-spec#114]
I'd like to request that we revisit how 'mute' functionality works on user objects. 

Currently, if a user mutes another user, posts from the muted user are only returned on endpoints where the muted message is (semi-)explicitly requested (e.g. get post by id).

However, that result would imply more of a 'block' to me rather than a 'mute'.  And I think both are necessary for best user experience.

A 'muted' user in my mind is someone I still want to see content from, but not necessarily all of it right now.  As an example, Xeni Jardin live-tweets a lot of her cancer treatment experiences in great, verbose detail.  She usually warns her followers before a particularly extensive live-tweet session on the topic, and if I'm following Xeni but not interested (n.b. I _am_ interested, this is just an example, please don't call me an insensitive clod) in her live-tweet marathon, I'd like to 'mute' her for now.  I'd probably like my client to show me that she posted a message, but maybe collapse it in my view.

A 'blocked' user is someone I never want to see content from - someone who is spammy, harrassing, generally annoying, or has a goatse avatar.  These posts shouldn't show up in any requests other than 'get post by id' requests.  Again, my opinion.

**Comment by shpakovski** [comment]
I would love to see an optional API argument allowing to mute a user for a period of time which would be set as a duration in seconds. As an example, “mute this user until the sport event is over in 3 hours”.

**Comment by orianmarx** [comment]
I agree that if the API is supporting mute functionality so as to keep mute state consistent across apps, it should support an optional timeout parameter at which point an unmute would occur. A client should be able to mute an already muted user with a new timeout without having to unmute first.

**Comment by jojolemon** [comment]
I support the idea of muting users as a separate thing from blocking them. It would be nice if the API supported it across apps...and for set periods of time which could be from a matter of minutes, to a fixed date, to indefinitely. I can see exchanges getting very heated, for example, during the period leading up the an election. I'd rather not have to deal with that. But, if I could mute someone until a date I could specify after the end of the election, that would be great. Likewise, if someone is "live-posting" an event I'm not interested in (a conference or a sporting event, for example), it would be nice to mute them for the duration of the event and not have to remember to go back into my settings and "un-mute" them.

In addition, I do not think info regarding mute-status should broadcast at all. If I decide to mute someone because they're "oversharing" at the moment, I don't want to hurt their feelings. I also don't want to know if anyone is tuning me out. Someone suggested on the alpha.app.net site that the system could broadcast something along the lines of follower-counts (without specifying who is doing the muting) which users could then use as part of a metrics on whether or not someone is worth following. Sadly, I suspect some users would take being muted as a sort of "badge of honor" and try to drive their "Mute Numbers" up by being as obnoxious and loutish as possible. No thank you.
